Microsoft.NET.SDK.IL doesn't work inside visual studio with multiple targetframeworks

When trying to build System.Runtime.Tests inside VS you would get the following error:

```
C:\Users\prgovi\.nuget\packages\microsoft.net.sdk.il\5.0.0-preview.4.20202.18\targets\Microsoft.NET.Sdk.IL.Common.targets(12,60): error MSB4226: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft.Common.CrossTargeting.targets" was not found. Also, tried to find "Microsoft.Common.CrossTargeting.targets" in the fallback search path(s) for $(MSBuildExtensionsPath) - "C:\Program Files (x86)\MSBuild" . These search paths are defined in "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\Bin\MSBuild.exe.Config". Confirm that the path in the <Import> declaration is correct, and that the file exists on disk in one of the search paths.
```

I took a look we're trying to import `Microsoft.Common.CrossTargeting.targets` from `$(MSBuildExtensionsPath)` which in DesignTimeBuild seems to not evaluate to the location where these targets are:
https://github.com/dotnet/runtime/blob/77a383275f1f5dc062a113102856718fe6e8b2ab/src/coreclr/src/.nuget/Microsoft.NET.Sdk.IL/targets/Microsoft.NET.Sdk.IL.Common.targets#L12

Changing it to `$(MSBuildToolsPath)` fixes the issue and works on both command line and DesignTimeBuild. 

Once that is fixed, then you would get an error:
```
C:\Users\prgovi\.nuget\packages\microsoft.net.sdk.il\5.0.0-preview.4.20202.18\targets\Microsoft.NET.Sdk.IL.targets(62,5): error : Package runtime.win-x86.microsoft.netcore.ilasm\5.0.0-preview.4.20202.18 was not restored
```

we're using `<_OSArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture)</_OSArchitecture>` to calculate the arch and the path of the `ilasm` and `ildasm` packages, but in DesignTimeBuild it seems to evaluate to `x86` even in a `x64` machine. 

https://github.com/dotnet/runtime/blob/77a383275f1f5dc062a113102856718fe6e8b2ab/src/coreclr/src/.nuget/Microsoft.NET.Sdk.IL/targets/Microsoft.NET.Sdk.IL.targets#L29

cc: @ViktorHofer @Anipik @ericstj

--- a/eng/Version.Details.xml
+++ b/eng/Version.Details.xml
@@ -158,13 +158,13 @@
       <Uri>https://github.com/dotnet/runtime</Uri>
       <Sha>cf64918877d98577363bb40d5eafac52beb80a79</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NETCore.ILAsm" Version="5.0.0-preview.4.20202.18">
+    <Dependency Name="Microsoft.NETCore.ILAsm" Version="5.0.0-preview.7.20316.4">
       <Uri>https://github.com/dotnet/runtime</Uri>
-      <Sha>0375524a91a47ca4db3ee1be548f74bab7e26e76</Sha>
+      <Sha>382dce3b4d205dd00f43564727d838404141e5e6</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NET.Sdk.IL" Version="5.0.0-preview.4.20202.18">
+    <Dependency Name="Microsoft.NET.Sdk.IL" Version="5.0.0-preview.7.20316.4">
       <Uri>https://github.com/dotnet/runtime</Uri>
-      <Sha>0375524a91a47ca4db3ee1be548f74bab7e26e76</Sha>
+      <Sha>382dce3b4d205dd00f43564727d838404141e5e6</Sha>
     </Dependency>
     <Dependency Name="System.Text.Json" Version="5.0.0-preview.4.20202.18">
       <Uri>https://github.com/dotnet/runtime</Uri>

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -68,7 +68,6 @@
     <MicrosoftExtensionsDependencyModelVersion>3.1.0</MicrosoftExtensionsDependencyModelVersion>
     <!-- CoreClr dependencies -->
     <MicrosoftNETCoreILAsmVersion>5.0.0-preview.7.20316.4</MicrosoftNETCoreILAsmVersion>
-    <MicrosoftNETSdkILVersion>5.0.0-preview.7.20316.4</MicrosoftNETSdkILVersion>
     <!-- Libraries dependencies -->
     <SystemTextJsonVersion>5.0.0-preview.4.20202.18</SystemTextJsonVersion>
     <runtimenativeSystemIOPortsVersion>5.0.0-alpha.1.19563.3</runtimenativeSystemIOPortsVersion>

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -67,8 +67,8 @@
     <MicrosoftNETCoreDotNetHostPolicyVersion>5.0.0-preview.4.20202.18</MicrosoftNETCoreDotNetHostPolicyVersion>
     <MicrosoftExtensionsDependencyModelVersion>3.1.0</MicrosoftExtensionsDependencyModelVersion>
     <!-- CoreClr dependencies -->
-    <MicrosoftNETCoreILAsmVersion>5.0.0-preview.4.20202.18</MicrosoftNETCoreILAsmVersion>
-    <MicrosoftNETSdkILVersion>5.0.0-preview.4.20202.18</MicrosoftNETSdkILVersion>
+    <MicrosoftNETCoreILAsmVersion>5.0.0-preview.7.20316.4</MicrosoftNETCoreILAsmVersion>
+    <MicrosoftNETSdkILVersion>5.0.0-preview.7.20316.4</MicrosoftNETSdkILVersion>
     <!-- Libraries dependencies -->
     <SystemTextJsonVersion>5.0.0-preview.4.20202.18</SystemTextJsonVersion>
     <runtimenativeSystemIOPortsVersion>5.0.0-alpha.1.19563.3</runtimenativeSystemIOPortsVersion>

--- a/global.json
+++ b/global.json
@@ -17,7 +17,7 @@
     "Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk": "5.0.0-beta.20309.1",
     "Microsoft.DotNet.Helix.Sdk": "5.0.0-beta.20309.1",
     "Microsoft.FIX-85B6-MERGE-9C38-CONFLICT": "1.0.0",
-    "Microsoft.NET.Sdk.IL": "5.0.0-preview.4.20202.18",
+    "Microsoft.NET.Sdk.IL": "5.0.0-preview.7.20316.4",
     "Microsoft.Build.NoTargets": "1.0.53",
     "Microsoft.Build.Traversal": "2.0.34"
   }

